pkg/resources/oauth_integration_for_custom_clients_acceptance_test.go

```diff
@@ -0,0 +1,667 @@
+					resource.TestCheckResourceAttrSet("snowflake_oauth_integration_for_custom_clients.test", "describe_output.0.oauth_authorization_endpoint.0.value"),
+					resource.TestCheckResourceAttrSet("snowflake_oauth_integration_for_custom_clients.test", "describe_output.0.oauth_token_endpoint.0.value"),
+					resource.TestCheckResourceAttrSet("snowflake_oauth_integration_for_custom_clients.test", "describe_output.0.oauth_allowed_authorization_endpoints.0.value"),
+					resource.TestCheckResourceAttrSet("snowflake_oauth_integration_for_custom_clients.test", "describe_output.0.oauth_allowed_token_endpoints.0.value"),
+				),
+			},
+			// import - complete
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_OauthIntegrationForCustomClients/complete"),
+				ConfigVariables: m(true),
+				ResourceName:    "snowflake_oauth_integration_for_custom_clients.test",
+				ImportState:     true,
+				ImportStateCheck: importchecks.ComposeAggregateImportStateCheck(
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "name", id.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_client_type", string(sdk.OauthSecurityIntegrationClientTypeConfidential)),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_redirect_uri", validUrl),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "enabled", "true"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_allow_non_tls_redirect_uri", "true"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_enforce_pkce", "true"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "pre_authorized_roles_list.#", "1"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "pre_authorized_roles_list.0", preAuthorizedRole.ID().Name()),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "blocked_roles_list.#", "3"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_issue_refresh_tokens", "true"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_refresh_token_validity", "86400"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "network_policy", sdk.NewAccountObjectIdentifier(networkPolicy.Name).Name()), // TODO(SNOW-999049): Fix during identifiers rework
+					// oauth_client_rsa_public_key and oauth_client_rsa_public_key_2 cannot be imported, because they're not available in SHOW and DESC outputs.
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "comment", comment),
+				),
+			},
+			// change externally
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_OauthIntegrationForCustomClients/complete"),
+				ConfigVariables: m(true),
+				PreConfig: func() {
+					acc.TestClient().SecurityIntegration.UpdateOauthForClients(t, sdk.NewAlterOauthForCustomClientsSecurityIntegrationRequest(id).WithUnset(
+						*sdk.NewOauthForCustomClientsIntegrationUnsetRequest().
+							WithEnabled(true).
```

Comment:
I wrote a few comments, I think the expectations are not correct based on the setup

Reply:
Hmm, the expectations where almost correct, but the address in the planchecks wasn't correct 🤦